adapters/repos/modules/modules.go

```diff
@@ -0,0 +1,135 @@
+//                           _       _
+// __      _____  __ ___   ___  __ _| |_ ___
+// \ \ /\ / / _ \/ _` \ \ / / |/ _` | __/ _ \
+//  \ V  V /  __/ (_| |\ V /| | (_| | ||  __/
+//   \_/\_/ \___|\__,_| \_/ |_|\__,_|\__\___|
+//
+//  Copyright © 2016 - 2020 SeMI Technologies B.V. All rights reserved.
+//
+//  CONTACT: hello@semi.technology
+//
+
+package modulestorage
+
+import (
+	"fmt"
+	"os"
+
+	"github.com/boltdb/bolt"
+	"github.com/pkg/errors"
+	"github.com/semi-technologies/weaviate/usecases/modules"
+	"github.com/sirupsen/logrus"
+)
+
+type Repo struct {
+	logger  logrus.FieldLogger
+	baseDir string
+	db      *bolt.DB
+}
+
+func NewRepo(baseDir string, logger logrus.FieldLogger) (*Repo, error) {
+	r := &Repo{
+		baseDir: baseDir,
+		logger:  logger,
+	}
+
+	err := r.init()
+	return r, err
+}
+
+func (r *Repo) DBPath() string {
+	return fmt.Sprintf("%s/modules.db", r.baseDir)
+}
+
+func (r *Repo) init() error {
+	if err := os.MkdirAll(r.baseDir, 0o777); err != nil {
+		return errors.Wrapf(err, "create root path directory at %s", r.baseDir)
+	}
+
+	boltdb, err := bolt.Open(r.DBPath(), 0o600, nil)
+	if err != nil {
+		return errors.Wrapf(err, "open bolt at %s", r.DBPath())
+	}
+
+	r.db = boltdb
+
+	return nil
+}
+
+type storageBucket struct {
+	bucketKey []byte
+	repo      *Repo
+}
+
+func (r *Repo) Storage(bucketName string) (modules.Storage, error) {
+	storage := &storageBucket{
+		bucketKey: []byte(bucketName),
+		repo:      r,
+	}
+
+	err := storage.init()
+	return storage, err
+}
+
+func (s *storageBucket) init() error {
+	return s.repo.db.Update(func(tx *bolt.Tx) error {
+		if _, err := tx.CreateBucketIfNotExists(s.bucketKey); err != nil {
+			return errors.Wrapf(err, "create module storage bucket '%s'",
+				string(s.bucketKey))
+		}
+
+		return nil
+	})
+}
+
+func (s *storageBucket) Put(key, value []byte) error {
+	return s.repo.db.Batch(func(tx *bolt.Tx) error {
+		b := tx.Bucket(s.bucketKey)
+		if b == nil {
+			return fmt.Errorf("no bucket for key %s found", string(s.bucketKey))
```

Comment:
I would use instead of `fmt.Error` the `errors.Wrapf` in this file

Reply:
what would I wrap here? It's a new error.

Reply:
I meant that to use concisely `errors` package for creating errors, don't mix `fmt` and `errors`.
You can replace this line of code (and following one where fmt.Errorf is being used) with `errors.Errorf("no bucket for key %s found", string(s.bucketKey))`

Reply:
ah, cool, didn't know `errors.Errorf()` existed. Yes, I'll update this.